code/game/objects/structures/urban.dm

```diff
@@ -392,6 +400,27 @@
 	icon_state = "exit"
 	layer = ABOVE_ALL_MOB_LAYER
 
+/obj/structure/sign/billboard
+	name = "commercial billboard"
+	desc = "A large and typically roadside billboard rented out for advertisement space."
+	icon = 'icons/obj/structure/urban/billboard.dmi'
+	icon_state = "board-l"
+	density = TRUE
+	layer = ABOVE_ALL_MOB_LAYER
+
+/obj/structure/sign/billboard/advert
+	name = "billboard advertisement"
+	desc = null
+	icon_state = "sign"
+	density = TRUE
+	layer = 4.6
+
+/obj/structure/sign/billboard/advert/Initialize(mapload)
+	. = ..()
+	cut_overlays()
+	icon_state = "sign[rand(1, 14)]"
```

Comment:
As commented on the other PR by Rusting.

Billboards should all be proper subtypes, so they can be placed in a map if a specific ad is needed.
That would allow them to get actual names and descs too, instead of just "billboard advertisement" with a random sprite. That would also allow in the future to have different billboards outside of Konyang.

And there should be a /obj/random/billboard or something that spawns a random billboard.
This is how potted plants work currently too, and you can look at /obj/random/pottedplant and the subtypes of /obj/structure/flora/pottedplant.